### PR TITLE
Keep `copyDependencyToDir()` from creating empty directories

### PR DESCRIPTION
**Pull request by gadenbuie**
When an `htmlDependency()` doesn't have any files, `copyDependencyToDir()` will still create an empty directory, as in the reprex below. This PR fixes this behavior by cleaning up the target directory when the dependency doesn't have any source files.

We still temporarily create the target directory and update the dependencies `$src$file` value, though, because rmarkdown will eventually call `makeDependencyRelative()` and we need `$src$file` to be a child of the document's `basedir` or an error is thrown.

````markdown
---
output: 
  html_document:
    self_contained: no
---

```{r echo=FALSE}
# create empty temp dir
tmpdir <- tempfile()
dir.create(tmpdir)

# add an htmldependency from that tempdir
htmltools::tagList(
  htmltools::htmlDependency(
    name = "empty",
    version = "9.9.9",
    src = tmpdir,
    head = "<script>alert('boo')</script>",
    all_files = FALSE
  )
)
```
````

### CODE REVIEW

**Review by cpsievert — APPROVED** [review]

File: R/html_dependency.R [hunk 1]
```diff
@@ -365,6 +366,13 @@ copyDependencyToDir <- function(dependency, outputDir, mustWork = TRUE) {
       )
     )
   }
+  if (!length(srcfiles)) {
```
Comment (schloerke) [inline R/html_dependency.R:369]: May we also move this if-block up to just after `files` is created? (~359)

```suggestion
  if (length(files) == 0) {
```

**Review by schloerke — APPROVED** [review]

**Comment by MoLi7** [comment]
I suppose this PR should fix the mysterious empty directories created after knitting a `xaringan` and `xaringanExtra` powered presentation, as described in this [StackOverflow](https://stackoverflow.com/questions/68498362/assets-folder-xaringan-gets-overwhelmed-with-new-folders) question.

Could someone merge this PR to implement @gadenbuie's proposal? Thank you!

**Comment by SNAnalyst** [comment]
I don't know it this adds any additional information, but I have found that this problem occurs when I use 

`xaringanextra::use_clipboard()`

(regardless of whether I add `htmltools::tagList()` after this. So, just `xaringanextra::use_clipboard()` creates this behavior by itself already.

File: R/html_dependency.R (hunk 1 above)
Comment (gadenbuie) [reply R/html_dependency.R]: Done in d945eb5!

**Comment by gadenbuie** [comment]
@cpsievert I made the change requested by @schloerke, this should be good to go now!

File: tests/testthat/test-deps.r [hunk 2]
```diff
@@ -332,3 +332,28 @@ test_that("html escaping is carried out correctly in script rendering", {
...
+test_that("copyDependencyToDir() doesn't create an empty directory", {
+  tmp_dep <- tempfile("dep")
+  dir.create(tmp_dep)
+  on.exit(unlink(tmp_dep))
+
+  tmp_rmd <- tempfile("rmd_files")
+  dir.create(tmp_rmd)
+  on.exit(unlink(tmp_rmd), add = TRUE)
+
+  empty <-   htmltools::htmlDependency(
```
Comment (cpsievert) [inline tests/testthat/test-deps.r:345]:
```suggestion
  empty <- htmltools::htmlDependency(
```

**Comment by cpsievert** [comment]
@gadenbuie thanks! Please also add a NEWS item and then I'll merge

File: R/html_dependency.R [hunk 3]
```diff
@@ -365,6 +374,7 @@ copyDependencyToDir <- function(dependency, outputDir, mustWork = TRUE) {
       )
     )
   }
+
```
Comment (cpsievert) [inline R/html_dependency.R:377]:
```suggestion
```

**Comment by cpsievert** [comment]
@schloerke any idea what's going wrong here? https://github.com/rstudio/htmltools/runs/4193559908?check_suite_focus=true#step:5:1270

**Comment by schloerke** [comment]
@cpsievert I have no clue. 😞 

It is extremely odd. It is tied to `r-lib/actions/setup-r-dependencies` in the line: `pak::pkg_deps("local::.", dependencies = TRUE)` when making the dependencies file for caching.

Why it is odd is that every other linux job works. I also busted the cache by incrementing the cache number, still no improvement. Also removed `extra-packages to be installed afterwards, still no improvement.

File: NEWS.md [hunk 4]
```diff
@@ -13,6 +13,8 @@
 
 * Closed #225: Added `tagInsertChildren()` to be able to insert child tag objects at a particular location. (#224)
 
+* `copyDependencyToDir()` no longer creates empty directories for dependencies that do not have any files. (@gadenbuie, #276)
```
Comment (cpsievert) [inline NEWS.md:16]: @gadenbuie would you mind rebasing/merging with `main` so this gets put in the right place?

**Pull request merged**